pkg/mq/interface.go

```diff
@@ -0,0 +1,163 @@
+package mq
+
+import "time"
+
+// Interface must be implemented for a message queuer
+type Interface interface {
+	// Connect to message queuer
+	Connect(timeout time.Duration) error
+
+	// Close connection
+	Close() error
+
+	// Publish
+	Publish(subject string, data []byte) error
+	PublishAsync(subject string, data []byte, ah AckHandler) (string, error)
+
+	// Subscribe
+	Subscribe(subject string, mh MsgHandler, opts ...SubscriptionOption) (Subscription, error)
```

Comment:
Do you plan to add [queue](https://github.com/nats-io/go-nats-streaming#queue-groups/) support in a separate PR?